### PR TITLE
Add PcPartPicker Presence

### LINKED ISSUE #3783: PcPartPicker | pcpartpicker.com
**Prerequisites and essential questions** 
- [x] Is it a popular site?
- [x] Is the website older than 2 months? 
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?** 
doechii#8001

**The Website is**

pcpartpicker.com

There should be different presences for if you are editing a list, browsing builds, or reading guides/
**If possible, please provide a logo for the service (512x512 minimum)** 
image

### PR DESCRIPTION
**Pull request by vahya**
Closes #3783

### CODE REVIEW

**Comment by vahya** [comment]
Proof of work
Screenshot 2021-06-25 164231
Screenshot 2021-06-25 164306
Screenshot 2021-06-25 164340
Screenshot 2021-06-25 164408
Screenshot 2021-06-25 164457
Screenshot 2021-06-25 164730
Screenshot 2021-06-25 164757
Screenshot 2021-06-25 164847
Screenshot 2021-06-25 164938
Screenshot 2021-06-25 164959

File: websites/P/PcPartPicker/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  "author": {
+    "name": "वाह्य",
+    "id": "819696394180100106"
+  },
+  "category": "other",
+  "logo": "https://i.imgur.com/4duoR2Q.png",
+  "thumbnail": "https://i.imgur.com/vWF3XN6.png",
+  "version": "1.0.0",
+  "color": "#c3a041",
+  "tags": ["other"],
```
Comment (Bas950) [inline websites/P/PcPartPicker/dist/metadata.json:13]: please add some tags related to pcs... not just other...

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/P/PcPartPicker/dist/metadata.json (hunk 1 above)
Comment (vahya) [reply websites/P/PcPartPicker/dist/metadata.json]: Sorry I was confused between tags and category

File: websites/P/PcPartPicker/presence.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        "tr.tr__total.tr__total--grandtotal > td.td__price"
+      );
+    presenceData.details =
+      build !== null && user !== null
+        ? `Viweing ${build.innerText} by ${user.innerText}`
+        : "";
+    presenceData.state = price !== null ? `Price: ${price.innerText}` : "";
+    presenceData.buttons = [
+      {
+        label: "Browse Build",
```
Comment (ririxidev) [inline websites/P/PcPartPicker/presence.ts:45]:
```suggestion
        label: "View Build",
```

File: websites/P/PcPartPicker/dist/metadata.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
+  "$schema": "https://schemas.premid.app/metadata/1.4",
+  "service": "PcPartPicker",
+  "author": {
+    "name": "वाह्य",
+    "id": "819696394180100106"
+  },
+  "category": "other",
+  "logo": "https://i.imgur.com/4duoR2Q.png",
```
Comment (ririxidev) [inline websites/P/PcPartPicker/dist/metadata.json:9]: min. 512x512 not 512x575 etc it isn't square like this.

File: websites/P/PcPartPicker/presence.ts [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    if (link !== null) {
+      presenceData.buttons = [
+        {
+          label: "View System",
+          url: link.value
+        }
+      ];
+    }
+  } else if (pathname.startsWith("/forums/"))
+    presenceData.details = "Browsing Forums";
```
Comment (ririxidev) [inline websites/P/PcPartPicker/presence.ts:102]: Add more things for forum

**Review by rxri — CHANGES_REQUESTED** [review]

File: websites/P/PcPartPicker/presence.ts [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+  else if (pathname.startsWith("/guide/")) {
+    const title: HTMLHeadingElement = document.querySelector(
+        "h1.pageTitle.guide__title"
+      ),
+      price: HTMLTableDataCellElement = document.querySelector(
+        "tr.tr__total.tr__total--final > td.td__price"
+      );
+    presenceData.details =
+      title !== null ? `Browsing ${title.innerText}` : "Browsing Guide";
+    presenceData.state = price !== null ? `Price: ${price.innerText}` : "";
```
Comment (Bas950) [inline websites/P/PcPartPicker/presence.ts:23]:
```suggestion
    if (price !== null)
       presenceData.state = `Price: ${price.innerText}`;
```

File: websites/P/PcPartPicker/presence.ts [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+      ),
+      user: HTMLAnchorElement = document.querySelector("div.user > a"),
+      price: HTMLTableDataCellElement = document.querySelector(
+        "tr.tr__total.tr__total--grandtotal > td.td__price"
+      );
+    presenceData.details =
+      build !== null && user !== null
+        ? `Viweing ${build.innerText} by ${user.innerText}`
+        : "";
+    presenceData.state = price !== null ? `Price: ${price.innerText}` : "";
```
Comment (Bas950) [inline websites/P/PcPartPicker/presence.ts:42]: same for these 2 (dont set states or details to "")

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/P/PcPartPicker/presence.ts (hunk 4 above)
Comment (vahya) [reply websites/P/PcPartPicker/presence.ts]: Should be better now

File: websites/P/PcPartPicker/presence.ts [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+      presenceData.state = `for ${product.innerText}`;
+    presenceData.buttons = [
+      {
+        label: "View Trends",
+        url: document.location.href
+      }
+    ];
+  } else if (pathname.startsWith("/builds/"))
+    presenceData.details = "Viewing Completed Builds";
+  if (presenceData.details === null) {
```
Comment (ririxidev) [inline websites/P/PcPartPicker/presence.ts:124]:
```suggestion
  if (!presenceData.details) {
```

File: websites/P/PcPartPicker/presence.ts [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+    const price: HTMLTableDataCellElement = document.querySelector(
+        "tr.tr__total.tr__total--final > td.td__price"
+      ),
+      link: HTMLInputElement = document.querySelector(
+        "div.actionBox.actionBox__permalink > input.text-input"
+      );
+    presenceData.details = "Building System";
+    presenceData.state =
+      price !== null ? `Price: ${price.innerText}` : "Price: $0";
+    if (link !== null) {
```
Comment (ririxidev) [inline websites/P/PcPartPicker/presence.ts:91]:
```suggestion
    if (link) {
```

File: websites/P/PcPartPicker/presence.ts [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+  } else if (pathname.startsWith("/list/")) {
+    const price: HTMLTableDataCellElement = document.querySelector(
+        "tr.tr__total.tr__total--final > td.td__price"
+      ),
+      link: HTMLInputElement = document.querySelector(
+        "div.actionBox.actionBox__permalink > input.text-input"
+      );
+    presenceData.details = "Building System";
+    presenceData.state =
+      price !== null ? `Price: ${price.innerText}` : "Price: $0";
```
Comment (ririxidev) [inline websites/P/PcPartPicker/presence.ts:90]:
```suggestion
      price ? `Price: ${price.innerText}` : "Price: $0";
```

File: websites/P/PcPartPicker/presence.ts [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+        label: "View Build",
+        url: document.location.href
+      }
+    ];
+  } else if (pathname.startsWith("/product/")) {
+    const productType: HTMLAnchorElement = document.querySelector(
+        "section.breadcrumb > ol.list-unstyled > li > a"
+      ),
+      productName: HTMLHeadingElement = document.querySelector("h1.pageTitle");
+    if (productType !== null)
```
Comment (ririxidev) [inline websites/P/PcPartPicker/presence.ts:52]:
```suggestion
    if (productType)
```

File: websites/P/PcPartPicker/presence.ts [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+      presenceData.details = `Looking at ${productType.innerText}`;
+    if (productName !== null) presenceData.state = productName.innerText;
```
Comment (ririxidev) [inline websites/P/PcPartPicker/presence.ts:54]:
```suggestion
    if (productName) presenceData.state = productName.innerText;
```

File: websites/P/PcPartPicker/presence.ts [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+  else if (pathname.startsWith("/guide/")) {
+    const title: HTMLHeadingElement = document.querySelector(
+        "h1.pageTitle.guide__title"
+      ),
+      price: HTMLTableDataCellElement = document.querySelector(
+        "tr.tr__total.tr__total--final > td.td__price"
+      );
+    presenceData.details =
+      title !== null ? `Browsing ${title.innerText}` : "Browsing Guide";
+    if (price !== null) presenceData.state = `Price: ${price.innerText}`;
```
Comment (ririxidev) [inline websites/P/PcPartPicker/presence.ts:23]:
```suggestion
    if (price) presenceData.state = `Price: ${price.innerText}`;
```

File: websites/P/PcPartPicker/presence.ts [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+  if (pathname === "/") presenceData.details = "Browsing Home Page";
+  else if (pathname.startsWith("/guide/")) {
+    const title: HTMLHeadingElement = document.querySelector(
+        "h1.pageTitle.guide__title"
+      ),
+      price: HTMLTableDataCellElement = document.querySelector(
+        "tr.tr__total.tr__total--final > td.td__price"
+      );
+    presenceData.details =
+      title !== null ? `Browsing ${title.innerText}` : "Browsing Guide";
```
Comment (ririxidev) [inline websites/P/PcPartPicker/presence.ts:22]:
```suggestion
      title ? `Browsing ${title.innerText}` : "Browsing Guide";
```

File: websites/P/PcPartPicker/presence.ts [hunk 14]
```diff
@@ -0,0 +1,128 @@
...
+    const build: HTMLHeadingElement = document.querySelector(
+        "h1.pageTitle.build__name"
+      ),
+      user: HTMLAnchorElement = document.querySelector("div.user > a"),
+      price: HTMLTableDataCellElement = document.querySelector(
+        "tr.tr__total.tr__total--grandtotal > td.td__price"
+      );
+    if (build !== null && user !== null)
+      presenceData.details = `Viweing ${build.innerText} by ${user.innerText}`;
+    if (price !== null) presenceData.state = `Price: ${price.innerText}`;
```
Comment (ririxidev) [inline websites/P/PcPartPicker/presence.ts:40]:
```suggestion
    if (price) presenceData.state = `Price: ${price.innerText}`;
```

File: websites/P/PcPartPicker/presence.ts [hunk 15]
```diff
@@ -0,0 +1,128 @@
...
+        {
+          label: "View System",
+          url: link.value
+        }
+      ];
+    }
+  } else if (pathname.startsWith("/forums/")) {
+    const topic: HTMLHeadingElement = document.querySelector("h1.pageTitle");
+    presenceData.details = "Browsing Forums";
+    if (topic !== null) {
```
Comment (ririxidev) [inline websites/P/PcPartPicker/presence.ts:102]:
```suggestion
    if (topic) {
```

File: websites/P/PcPartPicker/presence.ts [hunk 16]
```diff
@@ -0,0 +1,128 @@
...
+    ];
+  } else if (pathname.startsWith("/b/")) {
+    const build: HTMLHeadingElement = document.querySelector(
+        "h1.pageTitle.build__name"
+      ),
+      user: HTMLAnchorElement = document.querySelector("div.user > a"),
+      price: HTMLTableDataCellElement = document.querySelector(
+        "tr.tr__total.tr__total--grandtotal > td.td__price"
+      );
+    if (build !== null && user !== null)
```
Comment (ririxidev) [inline websites/P/PcPartPicker/presence.ts:38]:
```suggestion
    if (build && user)
```

**Review by rxri — CHANGES_REQUESTED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Review by rxri — APPROVED** [review]

**Pull request merged**